vitals/StateEstimator.java

```diff
@@ -0,0 +1,63 @@
+package vitals;
+
+public class StateEstimator {
+	
+	final float MAX_TEMPERATURE = 45;
+	final float MIN_TEMPERATURE = 0;
+	final float MAX_SOC= 80;
+	final float MIN_SOC= 20;
+	final float MAX_CHARGE_RATE= 0.8f;
+	
+	float temperature;
+	float stateofcharge;
+	float chargeRate;
+	
+	public StateEstimator() {
+		
+	}
+	public StateEstimator(float temperature, float soc, float chargeRate) {
+		this.temperature = temperature;
+		this.stateofcharge = soc;
+		this.chargeRate = chargeRate;
+	}
+	public boolean isBatteryOk() {
+		boolean status_of_temp=isTemperatureWithinRange(this.temperature);
+		boolean status_of_soc=isSocWithinRange(this.stateofcharge);
```

Comment:
Just give a thought-  why don't we pass required range check functions and value  as a parameter for is****** functions